File: SiteVisitsNetCore2020Web/ClientApp/src/app/models/PageTitle.ts
import { PageUrl } from "./PageUrl";

export class PageTitle {
    title: string;
  titleCompact: string;
  //pageUrl: PageUrl;
  constructor(options: {
    title: string,
    titleCompact: string,
    //pageUrl: PageUrl
  }) {
    if (options) {
      this.title = options.title;
      this.titleCompact = options.titleCompact;
      //this.pageUrl = options.pageUrl;
    }
  }
}

File: SiteVisitsNetCore2020Web/Models/PageTitle.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.ComponentModel.DataAnnotations;

namespace SiteVisitsNetCore2020Web.Models
{
    public class PageTitle
    {
        [Key]
        public string Title { get; set; }
        public string TitleCompact { get; set; }
        public DateTime FirstOccurrence { get; set; }
        public virtual PageUrl PageUrl { get; set; }
    }
}
